stl/src/primitives.hpp

```diff
@@ -8,132 +8,130 @@
 
 #include "awint.hpp"
 
-namespace Concurrency {
-    namespace details {
-        class __declspec(novtable) stl_critical_section_interface {
-        public:
-            virtual void lock()                     = 0;
-            virtual bool try_lock()                 = 0;
-            virtual bool try_lock_for(unsigned int) = 0;
-            virtual void unlock()                   = 0;
-            virtual void destroy()                  = 0;
-        };
-
-        class __declspec(novtable) stl_condition_variable_interface {
-        public:
-            virtual void wait(stl_critical_section_interface*)                   = 0;
-            virtual bool wait_for(stl_critical_section_interface*, unsigned int) = 0;
-            virtual void notify_one()                                            = 0;
-            virtual void notify_all()                                            = 0;
-            virtual void destroy()                                               = 0;
-        };
-
-        class stl_critical_section_win7 final : public stl_critical_section_interface {
-        public:
-            stl_critical_section_win7() {
-                InitializeSRWLock(&m_srw_lock);
-            }
+namespace Concurrency::details {

```

Comment:
Can you please revert the namespace style changes and the other whitespace changes. The only changes I like is the constexpr and the static -> dynamic cast.